x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.test.ts

```diff
@@ -10,19 +10,106 @@ import { PromiseReturnType } from '../../../typings/common';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths
+      );
+
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if service name and environment are given', () => {
+    it('excludes connections with different service name and environment', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const serviceName = 'opbeans-node';
+      const environment = 'production';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        serviceName,
+        environment
+      );
+      expect(
+        connections.every(
+          (conn) => conn.source['service.environment'] === environment
+        )
+      ).toBeTruthy();
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if service name is given', () => {
+    it('excludes connections with different service name', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const serviceName = 'opbeans-node';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        serviceName
+      );
+
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if environment is given', () => {
+    it('excludes connections with different environment', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const environment = 'testing';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        undefined,
+        environment
+      );
+
+      expect(
+        connections.every(
+          (conn) => conn.source['service.environment'] === environment
+        )
+      ).toBeTruthy();
+    });
+  });
+
+  describe('if environment is "not defined"', () => {
+    it('excludes connections with source environment set', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const environment = 'ENVIRONMENT_NOT_DEFINED';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        undefined,
+        environment
+      );
+
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if environment is "all" (missing)', () => {
+    it('includes both connections with and without source environment set', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths
+      );
+
+      expect(connections).toMatchSnapshot();
```

Comment:
Can we explicitly assert that the response contains the different cases:

```suggestion
      expect(
        connections.some(
          (conn) => conn.source['service.environment'] === 'production'
        )
      ).toBeTruthy();
      
      expect(
        connections.some(
          (conn) => conn.source['service.environment'] === 'development'
        )
      ).toBeTruthy();      
      
      expect(
        connections.some(
          (conn) => !!conn.source['service.environment']
        )
      ).toBeTruthy();            
      
```